retrieval/discovery/marathon/conversion.go

```diff
@@ -0,0 +1,76 @@
+package marathon
+
+import (
+	"fmt"
+	"strings"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+
+	"github.com/prometheus/prometheus/config"
+)
+
+// AppsToTargetGroups takes an array of Marathon apps and converts them into target groups.
+func AppsToTargetGroups(apps *AppList) map[string]*config.TargetGroup {
+	tgroups := map[string]*config.TargetGroup{}
+	for _, a := range apps.Apps {
+		if isValidApp(&a) {
+			group := createTargetGroup(&a)
+			tgroups[group.Source] = group
+		}
+	}
+	return tgroups
+}
+
+func createTargetGroup(app *App) *config.TargetGroup {
+	var (
+		targets = targetsForApp(app)
+		source  = targetGroupName(app)
+		job     = clientmodel.LabelValue(sanitizeName(app.ID))
+		image   = clientmodel.LabelValue(imageName(app))
```

Comment:
We decided for any information but the address to end up in meta labels, i.e. labels prefixed with `__meta_<sd_name>_` because we cannot generally assume a direct mapping into labels such as `job`. Setting other labels such as `image` could also collide with labels of actual time series.

Thus those should go to `__meta_marathon_app` and `__meta_marathon_image` and can then be relabeled into `job` and `image` as required. You can check the consul SD implementation for reference. Being flexible is more important than a generic config in this case.

Sorry for not bringing this up earlier. 
A reasonable exception, in my opinion, would be an SD type that explicitly has label sets assigned to targets, such as Kubernetes.